src/GameEngine/GameEngine.cpp

```diff
@@ -451,4 +500,100 @@ namespace GameEngine {
         *game_over = false;
     }
 
+    /*
+     * Tournament
+     */
+//        Tournament::Tournament(vector<Map*> *maps, vector<Player::Player*> *players, int *numGames, int *maxTurns) {
+//        this->maps = maps;
+//        this->players = players;
+//        this->numGames = numGames;
+//        this->maxTurns = maxTurns;
+//    }
+
+    Tournament::Tournament() {
+        map_indices = new vector<int>;
+        maps = new vector<Map*>;
+        players = new vector<Player::Player*>;
+        game_stats = new vector<vector<string *>*>;
+        num_maps = new int();
+        numGames = new int();
+        maxTurns = new int();
+    }
+
+    Tournament::~Tournament() {
+        delete numGames;
+        delete num_maps;
+        delete maxTurns;
+        delete players;
+        delete maps;
+    }
+    void Tournament::start() {
+        GameEngine *game_engine = GameEngine::instance();
+        for (int i = 0; i < map_indices->size(); i++) {
+            for (int j = 0; j < *numGames; j++) {
+                game_engine->set_tournament(*maxTurns);
+                //call new game
```

Comment:
really don't need that kind of comments that just rephrase the methods' name

Reply:
Lol I had todos that's why so I turned them into comments, I can remove if you want

Reply:
this has been fixed